cpp/src/lists/set_operations.cu

```diff
@@ -181,8 +179,10 @@ std::unique_ptr<column> intersect_distinct(lists_column_view const& lhs,
                                   stream,
                                   mr);
 
-  return null_count == 0 ? std::move(output)
-                         : cudf::detail::purge_nonempty_nulls(output->view(), stream, mr);
+  if (auto const output_cv = output->view(); cudf::detail::has_nonempty_nulls(output_cv, stream)) {
```

Comment:
Does checking `null_count == 0` as an extra early exit before calling `has_nonempty_null` help at all here, or is the perf difference negligible?

Reply:
`has_nonempty_nulls` also checks for it so we don't need to do it here.